### ISSUE TITLE
[Bug]: Greedy argmax can emit invalid token ids: zero-padded lm_head rows win when all real logits are negative

### ISSUE THREAD

**Issue by phact** [mirage-project/mirage#751]
Found on commit 5c28cc68 (branch `mpk`) while bringing MPK up on an RTX 5090, but the bug is architecture-independent.

## Summary

`demo/qwen3/demo.py` pads the lm_head weight from `vocab_size` (151,936 for Qwen3) to 153,600 rows with zeros. The padded rows therefore always produce logits of exactly 0. The argmax pipeline (`argmax_partial` --> `argmax_reduce`) scans the full padded width with no knowledge of the true vocab bound, so whenever every real logit is negative, a padding row wins and the emitted token id is ≥ vocab_size; invalid, and the decode trajectory is corrupted from there.

This is not a rare corner: the first decode position of a Qwen3 thinking-mode response is exactly such a step (HF transformers greedily picks `<think>`, id 151667, whose raw logit is negative). Any step whose top-1 logit is negative triggers it.

## Reproduction

```bash
python demo/qwen3/demo.py --use-mirage --model Qwen/Qwen3-0.6B --max-new-tokens 8 --save-tokens auto
python -c "import json; ids=json.load(open('outputs/qwen3/mpk_output.json'))['token_ids']; print(ids[0], 'INVALID' if ids[0]>=151936 else 'ok')"
```

Observed first generated id: 152243 (> 151,936). HF transformers greedy on the same prompt: 151667 (`<think>`).

## Root cause

Zero weight padding ⇒ padded logits ≡ 0; `0 > (all-negative real logits)` in the reduce. Weight-side fixes can't work because a padding row's logit is `w·h`, sign depends on the hidden state.

## Proposed fix (implemented locally, happy to PR)

Bound the reduce by the true vocab size via a defaulted template parameter (no ABI break, legacy instantiations unchanged):

- `tasks/ampere/argmax.cuh`: `template <..., long long VOCAB_SIZE = 0x7fffffffLL>`; skip candidates with `global_idx >= VOCAB_SIZE` in the reduce scan.
- `task_register.cc` `register_argmax_reduce_task`: accept an optional second param, emit the extra template arg when present.
- `persistent_kernel.py` `argmax_reduce_layer(..., vocab_size=None)`.
- demo passes `model.config.vocab_size`.

One second-order effect remains after this fix (the partial-argmax chunk that straddles the vocab boundary can still lose its real candidates to an in-chunk padding tie). Filing separately with its own fix.

With both fixes we get token-for-token greedy agreement with HF transformers for 47 tokens on the demo prompt, diverging only at a measured 0.25-logit near-tie.
